salt/modules/lxc.py

```diff
@@ -989,6 +1045,10 @@ def init(name,
     if users is None:
         users = []
     dusers = ['root']
+    # think to touch the default user generated by default ubuntu template which
+    # has a really unsecure password too ...
+    if __grains__['os'] in ['Ubuntu'] and 'ubuntu' not in users:
+        dusers.append('ubuntu')
```

Comment:
getent won't tell you if the container is ubuntu, just if there is an ubuntu user.

[This pull request](https://github.com/saltstack/salt/pull/20933) will address setting the password for Ubuntu containers. Ubuntu releases ship with lsb_release so this should be a reliable way of detecting Ubuntu hosts and automagically setting the password.